Epam/Java Basics_UpSkill Lab #1/Algorithmization/One_dimensional_arrays/src/com/androsor/array/HomeTask9.java

```diff
@@ -0,0 +1,57 @@
+package com.androsor.array;
+
+import java.util.Arrays;
+import java.util.Scanner;
+
+/**
+ * Find the most frequent number in an array of integers with n elements. If there are several such numbers,
+ * then determine the smallest of them.
+ */
+public class HomeTask9 {
+
+    public static void main (String[] args) {
+
+        int arrayWidth; // разрядность массива
+
+        System.out.print(" Введите разрядность массива arrayWidth = ");
+        arrayWidth = HomeTask1.inputParameter();
+
+        System.out.println(" Исходный массив: ");
+        System.out.println(" ------------------------------------------------------- ");
+        int[] myArray = HomeTask2.createArray((arrayWidth));
+        System.out.println(" " + Arrays.toString(myArray));
+        System.out.println(" ------------------------------------------------------- ");
+
+
+        int mostCommonNumberCurrent = myArray[0]; //текущий ниболее встречающийся элемент.
+        int counterLargest = 1; // Наибольшее число повторов.
+        int mostCommonNumber = 0; // Наиболее часто встречающийся элемент массива.
+
+        for (int i = 0; i < myArray.length - 1; i++) {
+
+            int counter = 1; // устанавливаем\сбрасываем счетчик.
+
+            for (int j = i + 1; j < myArray.length; j++) {
+                if (myArray[i] == myArray[j]) {
+                    counter++;
+                    mostCommonNumberCurrent = myArray[i];
```

Comment:
I'm not sure that it makes sense to assign the same value to a variable more than once.